### ISSUE TITLE
karyoploteR error for custom genome

### ISSUE THREAD

**Issue by tiramisutes** [bernatgel/karyoploteR#23]
Dear,
There is error when I use the karyoploteR for custom genome as follows. 
```
custom.genome <- toGRanges(data.frame(chr=c("Pn5"), start=c(1), end=c(50000)))
kp <- plotKaryotype(genome = custom.genome)

library(GenomicFeatures)
txdb <- makeTxDbFromGFF(file = "C:/Users/hope/Desktop/sample.GTF", format = "gtf")
all_genes <- genes(txdb)
all_genes

GRanges object with 5 ranges and 1 metadata column:
     seqnames      ranges strand |     gene_id
        <Rle>   <IRanges>  <Rle> | <character>
  g1      Pn5   1813-2068      + |          g1
  g2      Pn5 10171-10521      + |          g2
  g3      Pn5 12320-13084      + |          g3
  g4      Pn5 13385-13624      + |          g4
  g5      Pn5 13785-14425      + |          g5
  -------
  seqinfo: 1 sequence from an unspecified genome; no seqlengths
```
The error are coming when I use kpPlotDensity to show Gene Density and display gene use kpPlotMarkers function.
```
> kpPlotDensity(kp, data=all_genes,window.size = 50000)
Error in graphics::polygon(x = c(x.chr, rev(x.chr)), y = c(y0.chr, rev(y1.chr)),  : 
  plot.new has not been called yet
```
```
> kpPlotMarkers(kp, data=all_genes, labels=all_genes$gene_id)
Error in strwidth(rep("M", length(chr.labels)), units = "user") : 
  plot.new has not been called yet
```
Any help is much appreciated.
Thanks.

**Comment by bernatgel** [comment]
Hi @tiramisutes,

I'll take a look at it. 

Could it be that you closed the plot created by plotKaryotype before executing kpPlotDensity? all plottig functions assume that a plot is active and they never create a new one.

And could you tell if you are using Bioconductor 3.7 or the 3.8 released just yesterday? (the ouput from sessionInfo() would be perfect)

**Comment by tiramisutes** [comment]
Hi,
Thanks for your reply.
It's working when adjusting the scripts sequence. The kpPlotDensity function should be followed to plotKaryotype.

**Comment by bernatgel** [comment]
If you want to change the color of the standard cytobands you can use the `color.table` argument in `kpAddCytobands` to give a named vector of colors. 

Using custom named cytobands is not supported in the current version but it's in the TODO list.  Meanwhile, you can force it in a not very elegant way: 

  - The cytoband names MUST be in a column called *gieStain*, but they can actually be whatever you want
  -  Build a color table matching your names

Something like this will create a ranbow colored genome

```
library(karyoploteR)
kp <- plotKaryotype(ideogram.plotter = NULL)

cytobands <- unlist(tileGenome(tilewidth = 10e6, seqlengths = kp$chromosome.lengths))

cytobands$gieStain <- paste0("color", as.character(seq_along(cytobands)))

color.table <- rainbow(length(cytobands))
names(color.table) <- paste0("color", as.character(seq_along(cytobands)))

kp$cytobands <- cytobands #This is needed because kpAddCytobands does not accept a cytobands parameter
kpAddCytobands(kp, color.table = color.table)

```
If what you actually want is to highlight certain regions in the genome, you can combine `kpRect` with the new (released last week with Bioconductor 3.8) `data.panel="ideogram"` to plot anything ON the ideogram itself

Hope this helps

**Comment by mchlou** [comment]
@bernatgel I am really sorry but I can't quite figure this out. I am trying to color a specific cytoband and the color would depend on the cytoband names. I tried making a cytoband color dataframe that would match a color to a given cytoband name. I also did not catch why we need to paste color values under the gieStain column; particularly this line of code `cytobands$gieStain <- paste0("color", as.character(seq_along(cytobands)))`? What should i do if I want a particular color for a specific cytoband present in multiple chromosomes? Sorry if this comes as stupid.
